fcrepo-kernel-api/src/main/java/org/fcrepo/kernel/api/utils/RelaxedPropertiesHelper.java

```diff
@@ -0,0 +1,138 @@
+     * @return the date that should be set for the CREATED_BY or null if it should be
+     *         untouched
+     */
+    public static String getCreatedBy(final Iterable<Statement> statements) {
+        String username = null;
+        for (Statement added : statements) {
+            if (added.getPredicate().equals(CREATED_BY)) {
+                if (username == null) {
+                    username = added.getObject().asLiteral().getString();
+                } else {
+                    throw new MalformedRdfException(CREATED_BY + " may only appear once!");
+                }
+            }
+        }
+        return username;
+    }
+
+    /**
+     * Gets the modified date (if any) that was included within the statements.
+     * @param statements statements to consider
+     * @return the date that should be set for the LAST_MODIFIED_DATE or null if it should be
+     *         untouched
+     */
+    public static Calendar getModifiedDate(final Iterable<Statement> statements) {
+        Calendar cal = null;
+        for (Statement added : statements) {
+            if (added.getPredicate().equals(LAST_MODIFIED_DATE)) {
+                if (cal == null) {
+                    cal = RelaxedPropertiesHelper.parseExpectedXsdDateTimeValue(added.getObject());
+                } else {
+                    throw new MalformedRdfException(LAST_MODIFIED_DATE + " may only appear once!");
+                }
+            }
+        }
+        return cal;
+    }
+
+    /**
+     * Gets the modified by user (if any) that was included within the statements.
+     * @param statements statements to consider
+     * @return the date that should be set for the MODIFIED_BY or null if it should be
+     *         untouched
+     */
+    public static String getModifiedBy(final Iterable<Statement> statements) {
+        String username = null;
+        for (Statement added : statements) {
+            if (added.getPredicate().equals(LAST_MODIFIED_BY)) {
+                if (username == null) {
+                    username = added.getObject().asLiteral().getString();
+                } else {
+                    throw new MalformedRdfException(LAST_MODIFIED_BY + " may only appear once!");
+                }
+            }
+        }
+        return username;
+    }
+
+    /**
+     * Parses an RDFNode that is expected to be a literal of type xsd:dateTime into a Java Calendar
+     * object.
+     * @param node a node representing an xsd:dateTime literal
+     * @return a Calendar representation of the expressed dateTime
+     */
+    public static Calendar parseExpectedXsdDateTimeValue(final RDFNode node) {
+        final Object value = node.asLiteral().getValue();
+        if (value instanceof XSDDateTime) {
+            return ((XSDDateTime) value).asCalendar();
+        } else {
+            throw new IllegalArgumentException("Expected an xsd:dateTime!");
+        }
+    }
+
+    private RelaxedPropertiesHelper() {
```

Comment:
That sounds reasonable... I was surprised when checksytle told me it was required.